Build failure: `tic-80` on `*-darwin`

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

On Darwin/macOS:
```sh
nix build 'nixpkgs#tic-80'
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

x86_64: <https://hydra.nixos.org/build/314546451>; aarch64: <https://hydra.nixos.org/build/314546449>

### Relevant log output

```console
[…]
Running phase: buildPhase
build flags: -j4 SHELL=/nix/store/p0k9r5h8qs7220xdbdihhfgzwjcly70x-bash-5.3p3/bin/bash
[  0%] Building C object CMakeFiles/giflib.dir/vendor/giflib/dgif_lib.c.o
[  0%] Building C object CMakeFiles/blipbuf.dir/vendor/blip-buf/blip_buf.c.o
[  1%] Building C object CMakeFiles/luaapi.dir/src/api/luaapi.c.o
[  2%] Building C object CMakeFiles/png.dir/vendor/libpng/png.c.o
[  3%] Linking C static library lib/libblipbuf.a
[  4%] Building C object CMakeFiles/giflib.dir/vendor/giflib/egif_lib.c.o
[  4%] Building C object CMakeFiles/luaapi.dir/src/api/parse_note.c.o
[  4%] Building C object CMakeFiles/png.dir/vendor/libpng/pngerror.c.o
[  5%] Linking C static library lib/libluaapi.a
[  5%] Building C object CMakeFiles/giflib.dir/vendor/giflib/gif_err.c.o
[  6%] Building C object CMakeFiles/png.dir/vendor/libpng/pngget.c.o
[  7%] Building C object CMakeFiles/giflib.dir/vendor/giflib/gif_font.c.o
[  7%] Built target blipbuf
[  7%] Built target luaapi
[  7%] Building C object CMakeFiles/png.dir/vendor/libpng/pngmem.c.o
[  7%] Building C object CMakeFiles/giflib.dir/vendor/giflib/gif_hash.c.o
[  8%] Building C object CMakeFiles/lpeg.dir/vendor/lpeg/lpcap.c.o
[  9%] Building C object CMakeFiles/wren.dir/vendor/wren/src/optional/wren_opt_meta.c.o
[ 10%] Building C object CMakeFiles/png.dir/vendor/libpng/pngpread.c.o
[ 11%] Building C object CMakeFiles/giflib.dir/vendor/giflib/gifalloc.c.o
[ 11%] Building C object CMakeFiles/lpeg.dir/vendor/lpeg/lpcode.c.o
[ 11%] Building C object CMakeFiles/wren.dir/vendor/wren/src/optional/wren_opt_random.c.o
[ 11%] Building C object CMakeFiles/png.dir/vendor/libpng/pngread.c.o
[ 11%] Building C object CMakeFiles/giflib.dir/vendor/giflib/openbsd-reallocarray.c.o
[ 12%] Building C object CMakeFiles/wren.dir/vendor/wren/src/vm/wren_compiler.c.o
[ 13%] Building C object CMakeFiles/lpeg.dir/vendor/lpeg/lpprint.c.o
[ 14%] Linking C static library lib/libgiflib.a
[ 14%] Building C object CMakeFiles/lpeg.dir/vendor/lpeg/lptree.c.o
[ 14%] Built target giflib
[ 15%] Building C object CMakeFiles/png.dir/vendor/libpng/pngrio.c.o
CMakeFiles/mruby_vendor.dir/build.make:87: *** missing separator.  Stop.
make[1]: *** [CMakeFiles/Makefile2:533: CMakeFiles/mruby_vendor.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[…various other things building in parallel…]
make: *** [Makefile:156: all] Error 2
```

### Additional context

[The CMake file](https://github.com/nesbox/TIC-80/blob/main/cmake/mruby.cmake) that TIC-80 uses to build `mruby` calls the `xcrun` command to find the macOS SDK, using the deprecated [`exec_program`](https://cmake.org/cmake/help/latest/command/exec_program.html) CMake command to do so. Nixpkgs is using a fake `xcrun` command from [`xcbuild`](https://github.com/facebookarchive/xcbuild), which outputs `warning: unhandled Platform key FamilyDisplayName` before returning the SDK path. `exec_command` captures both stdout and stderr in the same variable, causing the warning to end up in the Makefile and breaking things.

I've just submitted nesbox/TIC-80#2869 to replace the `exec_command` call with [`execute_process`](https://cmake.org/cmake/help/latest/command/execute_process.html), the recommended replacement. `execute_process` handles stdout and stderr separately, so it doesn't have this problem.

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/bypd23xzpxkjwf6zkp8clfik9r7c8pql-nixpkgs/nixpkgs`

### Notify maintainers

@blinry

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ti/tic-80/package.nix
+++ b/pkgs/by-name/ti/tic-80/package.nix
@@ -21,7 +21,7 @@
 }:
 let
   # git rev-list HEAD --count
-  revision = "3042";
+  revision = "3049";
   year = "2025";
 in
 
@@ -29,18 +29,18 @@ stdenv.mkDerivation {
   pname = "tic-80";
   # use an untagged version until upstream tags a new version. We want
   # 'PREFER_SYSTEM_LIBRARIES', and without it tic-80 won't build
-  version = "1.1-unstable-2025-10-21";
+  version = "1.1-unstable-2025-12-30";
 
   src = fetchFromGitHub {
     owner = "nesbox";
     repo = "TIC-80";
-    rev = "a2c875f7275541e7724199ce8e504fb578b819a6";
+    rev = "881828910e77c799c1a6894cadc2d05a5c2f3f70";
     # TIC-80 vendors its dependencies as submodules. For the following dependencies,
     # there are no (or no compatible) packages in nixpkgs yet, so we use the vendored
     # ones as a fill-in: wasm, squirrel, pocketpy, argparse, naett,
     # sdlgpu, mruby.
     fetchSubmodules = true;
-    hash = "sha256-S/v1WHrvVndW4qHFWRJc3dcQkjw5vCU1/mN0PW9Wfkc=";
+    hash = "sha256-bqLVoLJpT62k9d+WZUgIu4ClcJNUvURB7j+NE/fzWNk=";
   };
 
   # TIC-80 tries to determine the revision part of the version using its Git history.

